CreateSymLinkFile.ps1

```diff
@@ -4,14 +4,54 @@ function Test-Admin {
     return $currentUser.IsInRole([Security.Principal.WindowsBuiltInRole]::Administrator)
 }
 
+function Confirm-Overwrite {
+    param (
+        $linkPath
+    )
+    Add-Type -AssemblyName PresentationFramework
+
+    # Presume that it's not a folder
+    $isDirectory = $false
+
+    if (-not [string]::IsNullOrWhiteSpace($linkPath)) {
+        # If the input box isn't empty, check if it's a valid path
+        if (Test-Path $linkPath -PathType Leaf) {
+            # if the path points to a file, it is not a folder
+            $isDirectory = $false
+        }
+        elseif (Test-Path $linkPath -PathType Container) {
+            # if the path points to a folder, it is a folder
+            $isDirectory = $true
+        }
+    }
+
+    $message = "A $(if ($isDirectory) { 'folder' } else { 'file' }) with the name '$(Split-Path $linkPath -Leaf)' already exists. Overwrite?"
+    $result = [System.Windows.MessageBox]::Show(
+        $message,
+        "Confirm Overwrite",
+        [System.Windows.MessageBoxButton]::YesNoCancel,
+        [System.Windows.MessageBoxImage]::Warning
+    )
+    return $result
+}
```

Comment:
this doesn't quite make sense.

the purpose of this script is to create symlinks to files. Unlike in linux where folders are files, in windows there's a distinct difference between files and containers (folders) so you can have an identically named file and folder in the same folder without causing a clash.

and a script asking you to overwrite a file-or-folder when it should just be creating files is overstepping it's boundaries and concerns.

the whole "is folder" idea should be scratched from here